frame/transaction-payment/src/payment.rs

```diff
@@ -98,7 +98,7 @@ where
 			WithdrawReasons::TRANSACTION_PAYMENT | WithdrawReasons::TIP
 		};
 
-		match C::withdraw(who, fee, withdraw_reason, ExistenceRequirement::KeepAlive) {
+		match C::withdraw(who, fee, withdraw_reason, ExistenceRequirement::AllowDeath) {
```

Comment:
Should ExistenceRequirement be pushed up into a `withdraw_fee` parameter? It fails the tx due to keep alive even though while this could reduce the native below the ED, there's enough sufficient asset in the account that it should still be ok to withdraw.

Reply:
Joe suggested KeepAlive should be right in both cases. Digging into balances pallet it's not taking into account sufficient assets when doing keep alive. I've made it less conservative and added tests (in assets pallet as I did not want to make balances pallet tests depend on assets pallet if I did not have to).